### PR TITLE
Np1979 resource service accepts and resturns Publication objects

### PR DESCRIPTION
**Pull request by axthosarouris**
## Note

In this PR some variables that should be named resource are called publication and vice versa. I will address this in a next PR, because if I change it here I will have many repeated collisions with  the already open PRs

### CODE REVIEW

File: publication-commons/build.gradle [hunk 1]
```diff
@@ -24,15 +27,14 @@ dependencies {
     api group: 'com.amazonaws', name: 'aws-java-sdk-lambda', version: project.ext.awsSdkVersion
     api group: 'com.amazonaws', name: 'aws-java-sdk-dynamodb', version: project.ext.awsSdkVersion
 
-    api group: 'com.github.BIBSYSDEV.nva-commons', name: 'identifiers', version: project.ext.nvaCommonsVersion
```
Comment (axthosarouris) [inline publication-commons/build.gradle:27]: we import the whole commons in line 6 (8 in the new version)

File: publication-commons/build.gradle [hunk 2]
```diff
@@ -24,15 +27,14 @@ dependencies {
...
 
     testImplementation project(":publication-testing")
     testImplementation('com.amazonaws:DynamoDBLocal:1.12.0') {
         exclude group: 'com.amazonaws', module: 'aws-java-sdk-dynamodb'
         exclude group: 'com.amazonaws', module: 'aws-java-sdk-core'
     }
 
-    api project (":storage-model")
+    testImplementation group: 'org.javers', name: 'javers-core', version: '5.14.0'
```
Comment (axthosarouris) [inline publication-commons/build.gradle:37]: for comparing java objects

File: publication-commons/src/main/java/no/unit/nva/publication/exception/InvalidPublicationException.java [hunk 3]
```diff
@@ -9,6 +9,10 @@
     public static final String ERROR_MESSAGE_TEMPLATE =
         "The Publication cannot be published because the following fields are not populated: ";
 
+    public InvalidPublicationException(String message) {
```
Comment (axthosarouris) [inline publication-commons/src/main/java/no/unit/nva/publication/exception/InvalidPublicationException.java:12]: Generic message is easier to maintain when fields are going to change name or location

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java [hunk 4]
```diff
@@ -93,68 +97,48 @@ public ResourceService(AmazonDynamoDB client, Clock clock) {
...
         newResource.setCreatedDate(clockForTimestamps.instant());
         TransactWriteItem[] transactionItems = transactionItemsForNewResourceInsertion(newResource);
         TransactWriteItemsRequest putRequest = newTransactWriteItemsRequest(transactionItems);
         sendTransactionWriteRequest(putRequest);
 
-        return fetchEventuallyConsistentResource(newResource);
+        return fetchEventuallyConsistentResource(newResource).toPublication();
     }
 
-    private Resource fetchEventuallyConsistentResource(Resource newResource) {
```
Comment (axthosarouris) [inline publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java:107]: private methods are placed under the public ones

File: publication-commons/build.gradle [hunk 5]
```diff
@@ -3,7 +3,10 @@ repositories {
 }
 
 dependencies {
+    api project(":storage-model")
```
Comment (axthosarouris) [inline publication-commons/build.gradle:6]: moved from the bottom of the dependencies. Dependency management will be done at the end in separate PR

**Review by brinxmat — APPROVED** [review]
